rtt/TaskContext.cpp

```diff
@@ -349,6 +349,8 @@ namespace RTT
             new_act = new SequentialActivity();
 #elif defined(ORO_ACT_DEFAULT_ACTIVITY)
             new_act = new Activity();
+#else
+            return false;
```

Comment:
This way you cannot clear the activity of a TaskContext anymore. Not sure if this use case would ever be required and it was broken anyway, but as having no activity (`our_act == NULL`) is the default state after construction if neither `ORO_ACT_DEFAULT_SEQUENTIAL` or `ORO_ACT_DEFAULT_ACTIVITY` is defined there is no reason why it should not be possible to also clear the activity using `setActivity(NULL)`. So instead of returning false here you could also check whether `new_act` is set in the remainder of the method and if not, only stop and destroy `our_act` if set.

Reply:
If no specific default activity type is given in the RTT configuration (neither `ORO_ACT_DEFAULT_SEQUENTIAL` nor `ORO_ACT_DEFAULT_ACTIVITY` is defined), the TaskContext could have no activity attached. I think this has been the default some time ago and is still a valid option.